flink-optimizer/src/main/java/org/apache/flink/optimizer/dag/TwoInputNode.java

```diff
@@ -427,6 +428,14 @@ public void computeInterestingPropertiesForInputs(CostEstimator estimator) {
 						}
 					}
 					
+					if (child1.getInputs().iterator().hasNext()) {
```

Comment:
This change is not necessary. The data distribution should be set within `GlobalProperties`. I'll point you to the correct places.